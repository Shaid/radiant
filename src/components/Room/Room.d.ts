export interface IState {
  currentRoom: number
  typingDone: boolean
}

export interface IProps {
}

export interface IDefaultProps {
}

export type PropsWithDefaults = IProps & IDefaultProps
