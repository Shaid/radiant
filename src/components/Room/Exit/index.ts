export { default } from './Exit'

// NOTE: uncomment this line to pass through named exports
// export * from './Exit'
