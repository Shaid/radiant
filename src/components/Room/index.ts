export { default } from './Room'

// NOTE: uncomment this line to pass through named exports
// export * from './Room'
