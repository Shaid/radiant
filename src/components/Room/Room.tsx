import React from 'react'

import styled from 'theme'

import Actor from 'components/Actor'
import Exits from './Exits'
import { IProps, IDefaultProps, IState } from './Room.d'

const Room = styled.div`
  display: flex;
  flex: 1 1 auto;
`

const RoomDescription = styled.div`
  flex: 1 1 auto;
`

const AreaName = styled.blockquote`
  font-size: 1.25rem;
  margin: 1rem 0;
`

const RoomName = styled.blockquote`
  font-size: 1.5rem;
  margin: 1rem 0;
`

const Actors = styled.div`
  flex: 1 1 auto;
  margin: 1rem 0;
`

// @todo Make room and it's travelTo() call use external props and state. Room should be from a store, etc.
const rooms = require('data/world/rooms.json')
const zones = require('data/world/zones.json')
const actors = require('data/actors/actors.json')

const getRandomActor = () => actors[Math.floor(Math.random() * actors.length)]


export default class extends React.PureComponent<IProps, IState> { // eslint-disable-line react/prefer-stateless-function
  static defaultProps: Partial<IDefaultProps> = {}

  constructor(props: IProps) {
    super(props)

    this.state = {
      currentRoom: 0
    }
  }

  travelTo = (destination: number) => {
    this.setState({
      currentRoom: destination
    })
  }

  render() { // eslint-disable-line class-methods-use-this
    const room = rooms[this.state.currentRoom]
    const zone = zones[room.zone]
    const sector = zone.sectors[room.sector]

    return (
      <Room>
        <RoomDescription>
          <AreaName>{zone.name}, <em>{zone.shortTitle}</em></AreaName>
          <RoomName><strong>{room.name}</strong>, {sector.name}</RoomName>
          {room.description.map((paragraph: string) => (<p key={paragraph}>{paragraph}</p>))}
          <Exits action={this.travelTo} exits={room.exits} />
          <Actors>
            <Actor actor={getRandomActor()} />
          </Actors>
        </RoomDescription>
      </Room>
    )
  }
}
