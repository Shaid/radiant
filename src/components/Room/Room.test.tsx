import React from 'react'
import { shallow } from 'enzyme'

import Room from '.'

describe('<Room />', () => {
  it('renders', () => {
    const wrapper = shallow(<Room />)
    expect(wrapper).toBeDefined()
  })
})
