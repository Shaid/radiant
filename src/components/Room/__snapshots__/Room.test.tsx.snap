// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Room /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Room__RoomDescription>
        <Room__AreaName>
          Villespont, 
          <em>
            the City of Bridges
          </em>
        </Room__AreaName>
        <Room__RoomName>
          <strong>
            Lower Arch Station
          </strong>
          , the demilitarised zone
        </Room__RoomName>
        <p>
          On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
        </p>
        <p>
          You feel the weight of the bridges above you.
        </p>
        <Room__Exits>
          <div>
            A bridge stretches away to the east.
          </div>
          <div>
            There is a closed door at the end of the platform.
          </div>
        </Room__Exits>
      </Room__RoomDescription>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Room__AreaName>
            Villespont, 
            <em>
              the City of Bridges
            </em>
          </Room__AreaName>,
          <Room__RoomName>
            <strong>
              Lower Arch Station
            </strong>
            , the demilitarised zone
          </Room__RoomName>,
          <p>
            On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
          </p>,
          <p>
            You feel the weight of the bridges above you.
          </p>,
          <Room__Exits>
            <div>
              A bridge stretches away to the east.
            </div>
            <div>
              There is a closed door at the end of the platform.
            </div>
          </Room__Exits>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Villespont, ",
              <em>
                the City of Bridges
              </em>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Villespont, ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "the City of Bridges",
              },
              "ref": null,
              "rendered": "the City of Bridges",
              "type": "em",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Lower Arch Station
              </strong>,
              ", the demilitarised zone",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lower Arch Station",
              },
              "ref": null,
              "rendered": "Lower Arch Station",
              "type": "strong",
            },
            ", the demilitarised zone",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
          },
          "ref": null,
          "rendered": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You feel the weight of the bridges above you.",
          },
          "ref": null,
          "rendered": "You feel the weight of the bridges above you.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                A bridge stretches away to the east.
              </div>,
              <div>
                There is a closed door at the end of the platform.
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A bridge stretches away to the east.",
              },
              "ref": null,
              "rendered": "A bridge stretches away to the east.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "There is a closed door at the end of the platform.",
              },
              "ref": null,
              "rendered": "There is a closed door at the end of the platform.",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Room__RoomDescription>
          <Room__AreaName>
            Villespont, 
            <em>
              the City of Bridges
            </em>
          </Room__AreaName>
          <Room__RoomName>
            <strong>
              Lower Arch Station
            </strong>
            , the demilitarised zone
          </Room__RoomName>
          <p>
            On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
          </p>
          <p>
            You feel the weight of the bridges above you.
          </p>
          <Room__Exits>
            <div>
              A bridge stretches away to the east.
            </div>
            <div>
              There is a closed door at the end of the platform.
            </div>
          </Room__Exits>
        </Room__RoomDescription>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Room__AreaName>
              Villespont, 
              <em>
                the City of Bridges
              </em>
            </Room__AreaName>,
            <Room__RoomName>
              <strong>
                Lower Arch Station
              </strong>
              , the demilitarised zone
            </Room__RoomName>,
            <p>
              On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
            </p>,
            <p>
              You feel the weight of the bridges above you.
            </p>,
            <Room__Exits>
              <div>
                A bridge stretches away to the east.
              </div>
              <div>
                There is a closed door at the end of the platform.
              </div>
            </Room__Exits>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Villespont, ",
                <em>
                  the City of Bridges
                </em>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Villespont, ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "the City of Bridges",
                },
                "ref": null,
                "rendered": "the City of Bridges",
                "type": "em",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <strong>
                  Lower Arch Station
                </strong>,
                ", the demilitarised zone",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lower Arch Station",
                },
                "ref": null,
                "rendered": "Lower Arch Station",
                "type": "strong",
              },
              ", the demilitarised zone",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
            },
            "ref": null,
            "rendered": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You feel the weight of the bridges above you.",
            },
            "ref": null,
            "rendered": "You feel the weight of the bridges above you.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div>
                  A bridge stretches away to the east.
                </div>,
                <div>
                  There is a closed door at the end of the platform.
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A bridge stretches away to the east.",
                },
                "ref": null,
                "rendered": "A bridge stretches away to the east.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "There is a closed door at the end of the platform.",
                },
                "ref": null,
                "rendered": "There is a closed door at the end of the platform.",
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
